wagtail/admin/forms/collections.py

```diff
@@ -18,11 +18,78 @@ class Meta:
         fields = ('restriction_type', 'password', 'groups')
 
 
+class SelectWithDisabledOptions(forms.Select):
+    """
+    Subclass of Django's select widget that allows disabling options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.disabled_values = ()
+
+    def create_option(self, name, value, *args, **kwargs):
+        option_dict = super().create_option(name, value, *args, **kwargs)
+        if value in self.disabled_values:
+            option_dict['attrs']['disabled'] = 'disabled'
+        return option_dict
+
+
+class CollectionChoiceField(forms.ModelChoiceField):
+    widget = SelectWithDisabledOptions
+
+    def __init__(self, *args, disabled_queryset=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._indentation_start_depth = 2
+        self._disabled_queryset = disabled_queryset
+        if disabled_queryset is not None:
+            self.widget.disabled_values = disabled_queryset.values_list(self.to_field_name or 'pk')
```

Comment:
Ah, we can actually simplify things and just do `self.disabled_queryset = disabled_queryset` - `_set_disabled_queryset()` will take care of updating the widget for us.

Reply:
Ohhh, I'm not used to using getters and setters, so I misread that line of yours as a typo. I'll fix it.

Reply:
Yeah, strong cases for getters and setters are pretty rare in my experience. They are practically non-existent in client work.